src/utils/cargo_metadata.rs

```diff
@@ -84,10 +83,26 @@ impl Package {
         self.library_target()
             .map(|target| self.normalize_package_name(&target.name))
     }
+
+    /// Deserialize metadata from lines.
+    pub fn from_lines<'a, T: AsRef<str>>(mut iter: impl Iterator<Item = T> + 'a) -> Result<Self> {
+        let metadata = if let (Some(serialized), None) = (iter.next(), iter.next()) {
```

Comment:
This seems unnecessarily generic - could you have it take `&[&str]` instead?

Reply:
Sure, I use it with [std::str::Lines](https://doc.rust-lang.org/std/str/struct.Lines.html) but I can collect it beforehand if you prefer that.

Reply:
Ah ok, that's fine then - I thought it was a slice to start.